### PR TITLE
PAE-124.1 - Exclude master course staff members from gradebook query.

### PR DESCRIPTION
**Pull request by Squirrel18**
## Description:

This PR improves https://github.com/proversity-org/edx-platform/pull/1132 by avoiding N+1 queries to the database and adds a generic function that can be used in subsequent changes.

## Notes:

Related PR: https://github.com/proversity-org/proversity-openedx-themes/pull/278

## Reviewers:

- [ ] @andrey-canon 
- [ ] @diegomillan

### CODE REVIEW

**Review by diegomillan — APPROVED** [review]

File: lms/djangoapps/ccx/utils.py [hunk 1]
```diff
@@ -444,3 +445,42 @@ def get_master_course_by_ccx_id(ccx_id):
         None or a CourseLocator instance.
     """
     return ccx_id.to_course_locator() if ccx_id else None
+
+
+def exclude_master_course_staff_users(users, course_key, model='User'):
```
Comment (andrey-canon) [inline lms/djangoapps/ccx/utils.py:450]: why did you include a second use case, if you are not using this, in which scenario will it be necessary ?

File: lms/djangoapps/ccx/utils.py [hunk 2]
```diff
@@ -444,3 +445,42 @@ def get_master_course_by_ccx_id(ccx_id):
...
+        If model value is not recognized will it return the users argument.
+    """
+    if not isinstance(course_key, CCXLocator):
+        return users
+
+    if model == 'User':
+        return users.exclude(
+            courseaccessrole__course_id=get_master_course_by_ccx_id(course_key),
+        )
+    elif model == 'CourseEnrollment':
```
Comment (andrey-canon) [inline lms/djangoapps/ccx/utils.py]: Unnecessary  `elif` after "return"

File: lms/djangoapps/ccx/utils.py (hunk 1 above)
Comment (Squirrel18) [reply lms/djangoapps/ccx/utils.py:450]: Next feature, PAE-125. @andrey-canon

File: lms/djangoapps/ccx/utils.py (hunk 2 above)
Comment (Squirrel18) [reply lms/djangoapps/ccx/utils.py]: Why? CourseEnrollement is not the default case... @andrey-canon

Comment (andrey-canon) [reply lms/djangoapps/ccx/utils.py]:
```suggestion
    if model == 'CourseEnrollment':
```
after return and raise statements the else condition is not necessary , the else is only necessary when the code execution keeps on

Comment (Squirrel18) [reply lms/djangoapps/ccx/utils.py]: Addressed. @andrey-canon

**Review by andrey-canon — APPROVED** [review]

**Pull request merged**